pkg/scheduler/core/generic_scheduler.go

```diff
@@ -204,18 +204,32 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 
 	trace.Step("Computing predicates")
 	startPredicateEvalTime := time.Now()
-	filteredNodes, failedPredicateMap, err := g.findNodesThatFit(pod, nodes)
+	predicateNodes, failedPredicateMap, err := g.findNodesThatFit(pod, nodes)
 	if err != nil {
 		return result, err
 	}
 
+	// TODO - Convert all built-in predicates to Plugin.
+	// In order to keep the backward compatiblity,
+	// RunFilterPlugins uses the result of all built-in predicates.
+	pc := framework.NewPluginContext() // FIXME after PreFilter implementation is committed
+	filteredNodes, failedMap := g.framework.RunFilterPlugins(pc, pod, predicateNodes)
```

Comment:
This will not really work. The problem is that `findNodesThatFit` stops looking for more feasible nodes after it finds a number of feasible nodes. This number changes depending on the size of cluster. For example, it could be 20% of the whole cluster. Let's assume that `findNodesThatFit` finds the 20% of the nodes and returns them. In that case, there may be many more feasible nodes in the cluster that are ignored by `findNodesThatFit`. Now, if we send those 20% to `Filter` plugins, they could filter all of them out. In such case, a pod would be considered unschedulable while there might be many more feasible nodes in the cluster. If we had converted all our predicates to `Filter` plugins, we could just call them in a very similar fashion that we call our `predicate` functions, but until then we need another solution.
There are a couple of solutions that immediately come to my mind:
1. Run Filter plugins before calling `findNodesThatFit` and pass only the nodes that pass `Filter` plugins to `findNodesThatFit`.
1. Run Filter plugins after `findNodesThatFit` (similar to what we have now in this PR), but pass all nodes to Filter plugins and then find intersection of nodes that passed Filter plugins and `findNodesThatFit`.

There might be more solutions. Between the two above, I would choose the first one for its simpler workflow.

Reply:
I will address it.